### DISCUSSION TITLE
Problem with file reader not picking up all files

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5940] [Q&A] [answered]
I have a file reader picking up files from an smb folder. The folder is not picking up all of the files in the directory. The file age is set to 50000 but the files are in there from days ago.

The file filter is a regex:  [A-Za-z].+Epic_Notes_Daily_[0-9]{8,}\..*

File names are like:  CNAMA_Epic_Notes_Daily_202309271340.txt

This passes the regex tester I tried it in. The files that have been picked up are of this same format, just with a different prefix: 

BNMG_Epic_Notes_Daily_202309281940.txt

The typical file size is 50-150 kb

Source queue is on (respond before processing)
Response: none

Any suggestions.

**Comment by jonbartels** [comment]
Regex looks good - https://regex101.com/r/j74bpo/2

What do you find if you search for the file name in your channel message history?

Any errors in `mirth.log`? Search for your channel ID in the logs.

**Comment by rutledj** [comment]
I'm not seeing any errors in the Mirth log. The channel doesn't even poll when you start it. The files are not in the mirth history either. It is like they don't exist.

**Comment by rutledj** [comment]
Update. There appears to be a bug with mirth and reading files from a folder that contains a large number of files. The folder contained 25k files but only 3500 match the regex criteria. I deleted all the other files and now the channel is reading them. We'll see if it reads all 3500 in before stopping again.

**Reply by jonbartels** [reply, marked as answer]
> There appears to be a bug with mirth and reading files from a folder that contains a large number of files

What bug did you find?

**Reply by rutledj** [reply]
The fact that it stops processing all the files in the folder that meet the file name filter criteria. It will do some then stop.  When I deleted all of the other files that don't meet the criteria, it processed all of the good one. So not sure how many files in a folder will cause this (I had 25k files). Maybe it is related somehow to using smb to connect?

**Reply by pacmano1** [reply]
Sorta sounds like a transient and/or network timeout problem.  Copy the files locally and try the same test to confirm.

**Reply by rutledj** [reply]
Unable to do that since mirth isn't allowed to connect to our PC's from the network.

**Reply by pacmano1** [reply]
huh?  locally = local to the mirth server.

**Reply by rutledj** [reply]
It is running on an appliance in our DMZ. No easy way to send that many files to a folder on the appliance.

**Comment by rjkroll** [comment]
Why even use RegEx at all?  If all the files you are looking for have 'Epic_Notes_Daily' in the name, then just search for `*Epic_Notes_Daily* `with a wildcard.  What is your source datatype set to?   Also, as a last resort, you could always try reading them via Javascript in a transformer to see if that way handles large amounts of files better.

**Comment by rutledj** [comment]
Sorry. Thought I attached a screenshot of the files. Some have Epic_Notes_Daily_Error, Epic_Notes_Daily_Validation in the name.

**Comment by rjkroll** [comment]
Can you screenshot your source/destination, etc or any relevant code so we can see what you are trying to do?
